code/modules/reagents/reagents/dispenser.dm

```diff
@@ -332,64 +332,67 @@
 
 /datum/reagent/acid/affect_touch(mob/living/carbon/M, alien, effect_multiplier) // This is the most interesting
-	if(ishuman(M))
-		var/mob/living/carbon/human/H = M
-		if(H.head)
-			if(H.head.unacidable)
-				to_chat(H, "<span class='danger'>Your [H.head] protects you from the acid.</span>")
-				remove_self(volume)
-				return
-			else if(volume > meltdose)
-				H << "<span class='danger'>Your [H.head] melts away!</span>"
-				qdel(H.head)
-				H.update_inv_head(1)
-				H.update_hair(1)
-				remove_self(meltdose)
-		if(volume <= 0)
-			return
-
-		if(H.wear_mask)
-			if(H.wear_mask.unacidable)
-				to_chat(H, "<span class='danger'>Your [H.wear_mask] protects you from the acid.</span>")
-				remove_self(volume)
-				return
-			else if(volume > meltdose)
-				H << "<span class='danger'>Your [H.wear_mask] melts away!</span>"
-				qdel(H.wear_mask)
-				H.update_inv_wear_mask(1)
-				H.update_hair(1)
-				remove_self(meltdose)
-		if(volume <= 0)
-			return
-
-		if(H.glasses)
-			if(H.glasses.unacidable)
-				H << "<span class='danger'>Your [H.glasses] partially protect you from the acid!</span>"
-				volume /= 2
-			else if(volume > meltdose)
-				H << "<span class='danger'>Your [H.glasses] melt away!</span>"
-				qdel(H.glasses)
-				H.update_inv_glasses(1)
-				remove_self(meltdose / 2)
-		if(volume <= 0)
-			return
-
-	if(volume < meltdose) // Not enough to melt anything
-		M.take_organ_damage(0, effect_multiplier * power * 0.2) //burn damage, since it causes chemical burns. Acid doesn't make bones shatter, like brute trauma would.
+	if(!ishuman(M))
+		M.apply_damage(volume * power * 0.2, BURN)
 		return
-	if(!M.unacidable && volume > 0)
-		if(ishuman(M) && volume >= meltdose)
-			var/mob/living/carbon/human/H = M
-			var/obj/item/organ/external/affecting = H.get_organ(BP_HEAD)
-			if(affecting)
-				if(affecting.take_damage(0, volume * power * 0.1))
-					H.UpdateDamageIcon()
-				if(prob(100 * volume / meltdose)) // Applies disfigurement
-					if (!(H.species && (H.species.flags & NO_PAIN)))
-						H.emote("scream")
-					H.status_flags |= DISFIGURED
-		else
-			M.take_organ_damage(0, volume * power * 0.1) // Balance. The damage is instant, so it's weaker. 10 units -> 5 damage, double for pacid. 120 units beaker could deal 60, but a) it's burn, which is not as dangerous, b) it's a one-use weapon, c) missing with it will splash it over the ground and d) clothes give some protection, so not everything will hit
+	var/mob/living/carbon/human/our_man = M
+	var/list/bodyparts = list(HEAD,UPPER_TORSO,LOWER_TORSO,ARM_LEFT,ARM_RIGHT,LEG_LEFT,LEG_RIGHT)
+	var/units_per_bodypart = volume / 7
+	var/list/obj/item/clothing/wearing_1 = list(
+		our_man.head,
+		our_man.glasses,
+		our_man.wear_suit,
+		our_man.shoes,
+		our_man.gloves
+	)
+	var/list/obj/item/clothing/wearing_2 = list(
+		our_man.wear_mask,
+		our_man.w_uniform,
+	)
+	message_admins("units per bodypart at [units_per_bodypart]")
```

Comment:
Ding ding ding, you left a debug message, -5 credits.

Reply:
day ruined.